django/contrib/gis/gdal/raster/source.py

```diff
@@ -250,7 +250,7 @@ def geotransform(self):
     @geotransform.setter
     def geotransform(self, values):
         "Set the geotransform for the data source."
-        if sum([isinstance(x, (int, float)) for x in values]) != 6:
+        if sum(isinstance(x, (int, float)) for x in values) != 6:
```

Comment:
this is just plain wrong, it allows through `[1,2,3,4,5,6,'foo']`. It should really be `len(values) == 6 and all(isinstance(x, (int, float) for x in values)`